### PR TITLE
NEO-2189 switch back to th for empty cell on header

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[Link to updated components in Deploy Preview](https://deploy-preview-457--neo-react-library-storybook.netlify.app/?path=/story/components-table--disabled-rows)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [ ] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

ADD PR SUMMARY (should be brief, one sentence if possible)

`@avaya-dux/dux-devs`:  
- try to change the background of dnd th cell along with checkbox th cell on tr:hover.
- switch back to use th for empty header cell.

## Summary by CodeRabbit

- **New Features**
  - Enhanced accessibility with improved drag-and-drop functionality in table headers.
  - Added new translation support for drag handle in tables.

- **Bug Fixes**
  - Disabled excessive logging for better performance in table components.

- **Style**
  - Improved table styling for better alignment, padding, and cursor interaction.

### CODE REVIEW

File: src/components/Table/TableComponents/TableHeader.tsx [hunk 1]
```diff
@@ -128,9 +128,14 @@ export const TableHeader = <T extends Record<string, any>>({
 		<thead>
 			<tr>
 				{canDrag && (
-					<td className="neo-table__dnd-th">
-						<div>&nbsp;</div>
-					</td>
+					<th className="neo-table__dnd-th">
+						<div
+							role="button"
+							aria-label={translations.dragHandle || "Drag handle"}
```
Comment (joe-s-avaya) [inline src/components/Table/TableComponents/TableHeader.tsx:134]: add `dragHandle` default text to the file `src/components/Table/helpers/default-data.ts` and then remove this `||`.
```suggestion
							aria-label={translations.dragHandle}
```

File: src/components/Table/TableComponents/TableHeader.tsx [hunk 2]
```diff
@@ -128,9 +128,14 @@ export const TableHeader = <T extends Record<string, any>>({
...
+						>
+							&nbsp;
+						</div>
+					</th>
```
Comment (joe-s-avaya) [inline src/components/Table/TableComponents/TableHeader.tsx:138]: This looks mostly good, but you will see two hover states if you hover from a vertical angle. If you hover horizontally, it works perfectly; it is _only_ the vertical hover that is problematic. 

Matt and Justin may say that this is fine for now, but check in with them.

**Review by joe-s-avaya — APPROVED** [review]
I have two comments that you should address, but this otherwise this looks good.

File: src/components/Table/TableComponents/TableHeader.tsx (hunk 1 above)
Comment (yangAtSpoken) [reply src/components/Table/TableComponents/TableHeader.tsx:134]: great!

File: src/components/Table/TableComponents/TableHeader.tsx (hunk 2 above)
Comment (yangAtSpoken) [reply src/components/Table/TableComponents/TableHeader.tsx:138]: Will tag them in next PR for overview of DND.

**Pull request merged**